src/EventProducerService.cpp

```diff
@@ -38,37 +38,65 @@ void EventProducerService::setProducedEvents()
     data[1] = lowByte(module_config->nodeNum);
     data[2] = 0;
     data[3] = i;
- 
-    byte index = i - 1;
+    
+    byte index = module_config->findExistingEventByEv(1, i);
+    if (index >= module_config->EE_MAX_EVENTS)  //event does not exist so creat default
+    {
+      index = module_config->findEventSpace();
+    }
+        
     module_config->writeEvent(index, data);
+    module_config->writeEventEV(index, 1, i);
     module_config->updateEvHashEntry(index);    
   }    
 }
 
+void EventProducerService::recreateDefaultEvent(byte evValue)
```

Comment:
I see code duplication here. Can you let setProducedEvents() call recreateDefaultEvent() in the loop. 
Does the condition there matter? For the conditions where is called (changing for Uninitialized to Normal), I expect the event table to be empty. I.e. findExistingEventByEv() won't find the events which means that findEventSpace() will be called, which is what recreateDefaultEvent() uses.
We can leave this for a later change if you prefer to isolate that refactoring and do more testing for just this change.